### DISCUSSION TITLE
TypeError: unhashable type: 'list'

### DISCUSSION THREAD

**Discussion by fipoucat** [xCDAT/xcdat#542] [2. Questions] [answered]
### Question criteria

- [X] I added a descriptive title here.
- [X] I searched the [xCDAT GitHub Discussions](https://github.com/xCDAT/xcdat/discussions) to find a similar question and didn't find it.
- [X] I searched the [xCDAT documentation](https://xcdat.readthedocs.io/en/stable/) and/or [Xarray documentation](https://docs.xarray.dev/en/stable/).

### Describe your question

I installed xcat and started testing the examples but noticed an error unhashable type.
```python
ds = xc.open_dataset(
    filepath,
    add_bounds=True,
    decode_times=True,
    center_times=True
)
```
```
Traceback (most recent call last):

  Cell In [3], line 3
    ds = xc.open_dataset(

  File ~/anaconda3/lib/python3.10/site-packages/xcdat/dataset.py:126 in open_dataset
    ds = _postprocess_dataset(ds, data_var, center_times, add_bounds, lon_orient)

  File ~/anaconda3/lib/python3.10/site-packages/xcdat/dataset.py:611 in _postprocess_dataset
    ds = ds.bounds.add_missing_bounds(axes=add_bounds)

  File ~/anaconda3/lib/python3.10/site-packages/xcdat/bounds.py:159 in add_missing_bounds
    for axis in axes:

TypeError: 'bool' object is not iterable
```
Any hint?

**Comment by jypeter** [comment]
@fipoucat Which version of `xcdat` are you using? The `0.5.0` version, or a more recent development version?

The release candidate discussion https://github.com/xCDAT/xcdat/discussions/517 mentions `add_bounds default value changed from add_bounds=True to add_bounds=["X","Y"]`. This indeed quite different from what is described in the [current open_dataset documentation](https://xcdat.readthedocs.io/en/stable/generated/xcdat.open_dataset.html)

You may want to remove `add_bounds=True` in your call to `xc.open_dataset`, or supply an explicit list of the axes where you want to add bounds

**Reply by tomvothecoder** [reply]
Thanks for the answer @jypeter, you are correct here. v0.5.0 updates `add_bounds=True` to `add_bounds=["X","Y"]`, with `True` being dropped as a valid type.

This brings up a good point that we should have probably added a deprecation warning and make `add_bounds=True` equivalent to `add_bounds=["X","Y"]`.  We'll be more careful about these breaking changes since the package is starting to mature.

**In the meantime, I'll add a patch so that users who upgrade from an older version to v0.5.0 don't run into this issue.** Just an FYI @xCDAT/core-developers.

**Reply by tomvothecoder** [reply]
Actually, v0.5.0 doesn't break and still uses `add_bounds=True`

https://github.com/xCDAT/xcdat/blob/d0a48920e84af3a80375dad64918ac1c9b7a0097/xcdat/dataset.py#L43

v0.6.0rc1 (a release candidate) introduces this change. We just need to update v0.6.0 to include a deprecation warning if users pass `True` still.

**Comment by fipoucat** [comment]
Thank you all, well noted.

**Reply by pochedls** [reply]
@fipoucat – were you using v0.5.0 (this will help us to understand if this is due to new changes or some other issue)? You can print the version with:

```
import xcdat as xc
print(xc.__version__)
```

**Comment by fipoucat** [comment]
Yes, I was using v0.5.0

**Reply by pochedls** [reply]
In tracing this down, it *seems* like you were using `v0.6.0rc1`, because you get the [`TypeError` on line 159 (for axis in axes)](https://github.com/xCDAT/xcdat/blob/v0.6.0rc1/xcdat/bounds.py#L159) of `bounds.py`, which doesn't seem to match [`v0.5.0`](https://github.com/xCDAT/xcdat/blob/main/xcdat/bounds.py#L159). I am a little confused about what is going on. 

Is the dataset you were using publicly available? I'm curious if the issue you had can be reproduced. 

...and...are you still having this issue? Or did changing the `add_bounds` argument address your issue (if so, please let us know how)?

**Comment by fipoucat** [comment]
I am using the example in the user guide

```python
import xcdat as xc
import  xesmf
# We import these packages specifically for plotting. It is not required to
use xcdat.
import matplotlib.pyplot as plt
import pandas as pd

filepath = "
https://esgf-data1.llnl.gov/thredds/dodsC/css03_data/CMIP6/CMIP/CSIRO/ACCESS-ESM1-5/historical/r10i1p1f1/Amon/tas/gn/v20200605/tas_Amon_ACCESS-ESM1-5_historical_r10i1p1f1_gn_185001-201412.nc
"
```

**Reply by pochedls** [reply, marked as answer]
@fipoucat – I think you are likely using version `0.6.0rc1` (this is a "release candidate," which means it is being tested; I think it will be released as v`0.6.0` soon). I can reproduce your issue with the release candidate version, but not with v`0.5`. 

You can: a) use version 0.5 for now, b) remove the `add_bounds=True` argument in the release candidate version, or c) wait until v0.6 is released (I think Tom is going to address the issue you raise here).

**Reply by tomvothecoder** [reply]
Thanks @pochedls. Yes I'm certain @fipoucat is using `xcdat=0.6.0rc1`. The installed version can be checked with this [comment](https://github.com/xCDAT/xcdat/discussions/542#discussioncomment-7028799) if we want to be sure.

>  b) remove the add_bounds=True argument in the release candidate version

Or replace `add_bounds=True` with `add_bounds=["X", "Y"]` if you want to add bounds if they are missing. "T" and "Z" are also valid axes.

> c) wait until v0.6 is released (I think Tom is going to address the issue you raise here).

I merged #546 to address this issue, which will be included in v0.6.0. `add_bounds=True` and `add_bounds=False` will be deprecated in v0.7.0.

**Comment by fipoucat** [comment]
Thank you, that sounds good.
